src/test/unit/gov/nist/javax/sip/parser/ims/PChargingVectorParserTest.java

```diff
@@ -0,0 +1,34 @@
+package test.unit.gov.nist.javax.sip.parser.ims;
+
+import gov.nist.javax.sip.parser.HeaderParser;
+import gov.nist.javax.sip.parser.ims.PChargingVectorParser;
+import test.unit.gov.nist.javax.sip.parser.ParserTestCase;
+
+public class PChargingVectorParserTest extends ParserTestCase {
+    
+    public void testParser() {
+        String[] preferredID =  {
+                "P-Charging-Vector: icid-value=value1;orig-ioi=value2;icid-generated-at=value3\n",
+
+                "P-Charging-Vector: icid-value=[2a02:ed0:1000:3033::249];orig-ioi=1223abc4;icid-generated-at=[2a02:ed0:1000:3033::249]\n",
+                
+                };
+        
+        super.testParser(PChargingVectorParser.class, preferredID);
+    }
+
+    public void testInvalidHeaderParam() {
+        
+        String preferredID =  "P-Charging-Vector: icid-value=[2a02:ed0:1000:3033::249k];orig-ioi=1223abc4;icid-generated-at=[2a02:ed0:1000:3033::249]\n";
+        
+        try {                  
+           HeaderParser hp = createParser(PChargingVectorParser.class, preferredID);
+           hp.parse();
+           fail("Header:"+ preferredID + " cannot be parse, wrong in pattern");
```

Comment:
No, If invalid IPV6 parsed then the test case shall be failed because the function "fail()" shall be called. Currently the testcase is passed so that the invalid IPv6 was not parsed.